### ISSUE TITLE
Make `cfg_has_atomic_x` public

### ISSUE THREAD

**Issue by Finomnis** [taiki-e/portable-atomic#133]
Quote from the library:

> // Not public API. (please submit an issue if you want this to be public API)

# Rationale

I have a library crate that wants to 
```rust
impl MyTrait for AtomicU8 { /* ... */ }
impl MyTrait for AtomicU16 { /* ... */ }
impl MyTrait for AtomicU32 { /* ... */ }
impl MyTrait for AtomicU64 { /* ... */ }
```

The problem is that one of the target architectures does not support AtomicU64 by default, and as a lib I shouldn't enable any features of `portable-atomic`. So I get the compiler error that `AtomicU64` does not exist.

I need a `cfg` detection of whether or not a specific atomic exists, so I can disable `impl`s for unavailable atomic types.

**Comment by taiki-e** [comment]
Thanks for the request. I'm open to making them public, but I have one question.

> one of the target architectures does not support AtomicU64 by default

At first glance, your use case seems to always require AtomicU64, why have you disabled the *enabled by default* `fallback` feature? If it is enabled, AtomicU64 should be provided except for one exception (mipsel-psx).

**Comment by Finomnis** [comment]
It's more in the line of : my function can receive any atomic integer. And I'm trying to model this by introducing a trait for it. So I don't "require all atomics", instead I would like to "support all existing atomics"

**Comment by taiki-e** [comment]
> I did try the "fallback" feature, but `AtomicU64` still wasn't available at thumbv6m-none-eabi.

Your code used fetch_add (which is not available in thumbv6m by default), so I was wondering if you enabled critical-section or single-core related features, otherwise AtomicU64 is indeed not provided even with fallback feature enabled.

**Comment by Finomnis** [comment]
Thanks <3

**Comment by taiki-e** [comment]
Published in 1.6.0.
